addons/mrp/models/stock_move.py

```diff
@@ -359,12 +359,14 @@ def _key_assign_picking(self):
     def _prepare_merge_moves_distinct_fields(self):
         distinct_fields = super()._prepare_merge_moves_distinct_fields()
         distinct_fields.append('created_production_id')
+        distinct_fields.append('bom_line_id')
```

Comment:
@odoo/rd-logistics This comment is for both `_prepare_merge_moves_distinct_fields` and `_prepare_merge_move_sort_method`. We believe that `bom_line_id` is an appropriate qualifier to identify a move as distinct from others.

When we sell a kit with (C1 and C2 as components) and at the same order, we add C1, the created picking will only contain 2 stock moves (C1@2, C2@1). We believe that it should be 3 (C1@1, C2@1 and C1@1) stock moves since the first C1 move is for the kit and the second C1 is for the separate C1 in the order. Thus, we propose adding `bom_line_id` as another qualifier to distinguish a move from the others. But we are not sure. Can you confirm?

Reply:
@caburj Agreed :)